components/dashboard/src/components/DropDown2.tsx

```diff
@@ -201,7 +203,13 @@ export const DropDown2: FunctionComponent<DropDown2Props> = ({
                         </div>
                     )}
                     <ul className="max-h-60 overflow-auto">
-                        {filteredOptions.length > 0 ? (
+                        {loading && (
+                            <div className="flex-col space-y-2 animate-pulse">
```

Comment:
**thought**: It feels like we overuse skeleton loaders in this form—maybe we could remove them from some elements or make sure we have everything in place during dashboard load, not when landing in this page. 🤷 

<img width="622" alt="Screenshot 2023-09-14 at 19 48 31" src="https://github.com/gitpod-io/gitpod/assets/120486/648c5ff0-9a35-4554-95ca-86fcfb3581d5">

Reply:
Yah... I think that's generally better - but would require quite a change in how we manage loading data for routes. I think we'd need to be much more explicit about what data is required for what page to render. I'm open to a differet way to show loading state here. There will be a couple scenarios where we would want to indicate loading here:

- Initial page load. We need to load your suggested repos and projects. This is needed both to show the list of options, but also to render the selected value correctly (i.e is it a project, or just a repo). This would be more of a blocking loading state as we can't render until we have this. We use skeletons here to indicate that atm.
- Future: If we incrementally search your git providers for matching repos as you type so we can find matches across your orgs too (not just top personal repos). This could be a non-blocking loading state of the component.

I do think once we replace the ide/class selectors with lighter weight UI, it will at least tone down how many skeletons load on this page.

Reply:
Agree, all valid points. Let's leave this as is for now and get back to it if users start finding this confusing.
Thanks, @selfcontained.